ftw/testing/freezer.py

```diff
@@ -63,12 +67,49 @@ def now(tz=None):
         expect(time_class()).call(lambda: new_time).count(0, None)
 
         self.mocker.replay()
+        self.enabled = True
+        transaction.get().addBeforeCommitHook(self.transaction_before_commit_hook)
         return self
 
     def __exit__(self, exc_type, exc_value, traceback):
+        datetime_mock_class = datetime
         self.mocker.restore()
         self.mocker.verify()
 
+        # The labix mocker does only replace pointers to the mock-class which
+        # are in dicts.
+        # This does not change the class of instances of our mock-class.
+        # In order to be able to commit while the time is freezed,
+        # we need to replace those classes manually.
+        # Since we cannot patch the class of existing instances,
+        # we need to construct new instances with the real datetime class and
+        # replace all mock instances.
+        for referrer in gc.get_referrers(datetime_mock_class):
+            if not isinstance(referrer, datetime_mock_class):
+                continue
+            if getattr(referrer, '__class__') != datetime_mock_class:
+                continue
+            replacement = datetime(*referrer.timetuple()[:6] + (referrer.microsecond,))
```

Comment:
If you want to serialize/deserialize Python datetimes with microsecond precision, I believe something along those lines is the only way - at least I couldn't find a cleaner way. For example, this looks somewhat cleaner, but truncates microseconds:
```
>>> dt
datetime.datetime(2015, 9, 15, 13, 47, 29, 453837)
>>> datetime.fromtimestamp(time.mktime(dt.timetuple()))
datetime.datetime(2015, 9, 15, 13, 47, 29)
```